### DISCUSSION TITLE
Differences in magnE when running simulation with centre, pos_ydir, and distance versus matsimnibs

### DISCUSSION THREAD

**Discussion by wys5188** [simnibs/simnibs#326] [8. Simulation workflows] [answered]
Dear Experts,
We are able to smoothly convert between centre/pos_ydir/distance <--> matsimnibs in the forward direction using calc_matsimnibs (https://github.com/simnibs/simnibs/discussions/324) and in the backwards direction using the formula Torge provided (https://github.com/simnibs/simnibs/discussions/318). However, when we do a sanity check to compare the magnE files in each case to ensure their equivalency, the field values and distribution differ ever so slightly. Do you know why this is the case? Is the difference meaningful or within the acceptable error of TMS?

This is what the magnE looks like when we use centre/pos_ydir/distance:
<img width="919" alt="image" src="https://github.com/simnibs/simnibs/assets/17935441/8a6b89cb-bff5-483a-9304-5ff542e10450">

This is what the magnE looks like when we use matsimnibs:
<img width="923" alt="image" src="https://github.com/simnibs/simnibs/assets/17935441/af1c7615-f6c6-4c65-b07e-5780928c9342">

You can see that the max value for the first case is 1.43, and for the second it is 1.42. The distribution also shifts slightly.

Thank you,
Wendy

**Comment by TorgeW** [comment, marked as answer]
This is to be expected. 
It is a lossy transformation from matsimnibs to center + pos_ydir. That is because pos_ydir is projected to the skin surface.

The right solution would just be to use the matsimnibs ones created or externally store your original center + pos_ydir. 

You can read the code of how the pos_ydir is used here:
https://github.com/simnibs/simnibs/blob/d045e8ccda6b349cbec639563af32dac69e95909/simnibs/mesh_tools/mesh_io.py#L1866
